pkg/trigger/discovery_handler.go

```diff
@@ -104,48 +55,52 @@ func (mt *MigrationTrigger) cleanMigrations(ctx context.Context, r metav1.APIRes
 	return nil
 }
 
-func (mt *MigrationTrigger) launchMigration(ctx context.Context, resource migrationv1alpha1.GroupVersionResource) error {
+func (mt *MigrationTrigger) launchMigration(ctx context.Context, gvr schema.GroupVersionResource) error {
 	m := &migrationv1alpha1.StorageVersionMigration{
 		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: storageStateName(resource) + "-",
```

Comment:
We need to garbage collects the old storageVersionMigration. `cleanMigrations` won't GC the old ones. We can do this in a follow-up.